ibis/pandas/core.py

```diff
@@ -384,3 +466,47 @@ def execute_and_reset(
     elif isinstance(result, pd.Series):
         return result.reset_index(drop=True)
     return result
+
+
+compute_time_context = Dispatcher(
+    'compute_time_context',
+    doc="""\
+
+Compute time context for a node in execution
+
+Notes
+-----
+For a given node, return with a list of timecontext that are going to be
+passed to its children nodes.
+time context is useful when data is not uniquely defined by op tree. e.g.
+a TableExpr can represent the query select count(a) from table, but the
+result of that is different with time context (20190101, 20200101) vs
+(20200101, 20210101), because what data is in "table" also depends on the
+time context. And such context may not be global for all nodes. Each node
+may have its own context. compute_time_context computes attributes that
+are going to be used in executeion and passes these attributes to children
+nodes.
+
+Param:
+timecontext : Optional[Tuple[pd.Timestamp, pd.Timestamp]]
+    begin and end time context needed for execution
+
+Return:
+List[Optional[Tuple[pd.Timestamp, pd.Timestamp]]]
```

Comment:
i think we should define the type ```CONTEXT=Tuple[pd.Timestamp, pd.Timestamp]``` somewhere and use it

Reply:
Same as above, defined in `typing.py`